<launch>
    <arg name="slam_body_id" default="green/base_footprint"/>
    <arg name="slam_odom_id" default="green/odom"/>
    <arg name="slam_wheel_left" default="wheel_left_joint"/>
    <arg name="slam_wheel_right" default="wheel_right_joint"/>

    <arg name="control_rviz_cfg" default="$(find-pkg-share nuslam)/config/slam_rviz_config.rviz"/>
    <include file="$(find-pkg-share nuturtle_control)/launch/start_robot.launch.xml">
        <arg name="control_rviz_cfg" value="$(var control_rviz_cfg)"/>
    </include>

    <include file="$(find-pkg-share nuturtle_description)/launch/load_one.launch.py">
        <arg name="color" value="green"/>
        <arg name="use_rviz" value="false"/>
        <arg name="use_jsp" value="true"/>
    </include>

    <node pkg="nuslam" exec="slam" name="slam">
        <param from="$(find-pkg-share nuturtle_description)/config/diff_params.yaml"/>
        <param name="slam_body_id" value="$(var slam_body_id)"/>
        <param name="slam_odom_id" value="$(var slam_odom_id)"/>
        <param name="wheel_left" value="$(var slam_wheel_left)"/>
        <param name="wheel_right" value="$(var slam_wheel_right)"/>
        <remap from="/joint_states" to="blue/joint_states"/>
    </node>

    <node pkg="tf2_ros"
        exec="static_transform_publisher"
        name="nusim_world_to_map"
        args="--x 0
            --y 0
            --z 0
            --qx 0
            --qy 0
            --qz 0
            --qw 1
            --frame-id nusim/world
            --child-frame-id map"
    />

</launch>
